addon/mixins/jsonapi-serializer.js

```diff
@@ -0,0 +1,31 @@
+import Ember from 'ember';
+import parentSerializerMixin from 'ember-data-partial-model/mixins/serializer';
+const { Mixin } = Ember;
+
+export default Mixin.create(parentSerializerMixin, {
+  serializeBelongsTo: function(snapshot, json, relationship) {
+    if (relationship.options.isPartialExtension) {
+      return null;
+    } else {
+      return this._super(...arguments);
+    }
+  },
+
+  _normalizePartialRelationship: function(modelClass, hash, descriptor, context) {
+    hash.relationships = hash.relationships || {};
+    hash.relationships[descriptor.key] = {
+      data: {
+        type: descriptor.type,
+        id: context.extractId(modelClass, hash)
+      }
+    };
+  },
+
+  _copyAttributesFromPartialToParent: function(ourHash, partialData) {
```

Comment:
The naming of `ourHash` feels odd, what about `resourceHash`? but also `partialHash`